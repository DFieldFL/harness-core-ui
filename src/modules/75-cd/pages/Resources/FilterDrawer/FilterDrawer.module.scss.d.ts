/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly filterDrawer: string
  readonly secondaryBtn: string
}
export default styles
