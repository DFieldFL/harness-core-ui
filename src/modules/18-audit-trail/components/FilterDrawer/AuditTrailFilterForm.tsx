import { FormInput, MultiSelectOption } from '@wings-software/uicore'
import React, { useState } from 'react'
import { useParams } from 'react-router-dom'
import type { FormikProps } from 'formik'
import { useMutateAsGet } from '@common/hooks'
import { StringKeys, useStrings } from 'framework/strings'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { useGetUsers, useGetOrganizationAggregateDTOList } from 'services/cd-ng'
import { UserItemRenderer, UserTagRenderer } from '@rbac/utils/utils'
import { actionToLabelMap, getResourceTypeForMultiselect, moduleToLabelMap } from '../../utils/RequestUtil'
import type { AuditTrailFormType } from './FilterDrawer'

interface AuditTrailFormProps {
  formikProps: FormikProps<AuditTrailFormType>
}

const AuditTrailFilterForm: React.FC<AuditTrailFormProps> = () => {
  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
  const [userQuery, setUserQuery] = useState<string>()
  const [orgQuery, setOrgQuery] = useState<string>()
  // const [projectsQuery, setProjectsQuery] = useState<string>()

  const { data: userData } = useMutateAsGet(useGetUsers, {
    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier },
    body: {
      searchTerm: userQuery
    },
    debounce: 300
  })

  const { data } = useGetOrganizationAggregateDTOList({
    queryParams: { accountIdentifier: accountId, searchTerm: orgQuery },
    debounce: 300
  })

  const { getString } = useStrings()
  const users =
    userData?.data?.content?.map(user => {
      return {
        label: user.name || user.email,
        value: user.email
      }
    }) || []

  const organizations =
    data?.data?.content?.map(org => ({
      label: org.organizationResponse.organization.name,
      value: org.organizationResponse.organization.identifier
    })) || []

  // const getOrgs = (): string[] => {
  //   if (orgIdentifier) {
  //     return [orgIdentifier]
  //   }

  //   return props.formikProps.values.organizations?.map(org => org.value as string) || []
  // }

  // const { data: projectData, refetch: refetchProjectList } = useGetProjectAggregateDTOList({
  //   queryParams: {
  //     accountIdentifier: accountId,
  //     searchTerm: projectsQuery,
  //     // orgIdentifiers: getOrgs(),
  //     pageSize: 10
  //   },
  //   debounce: 300
  // })

  const getOptionsForMultiSelect = (map: Record<any, StringKeys>): MultiSelectOption[] => {
    return Object.keys(map).map(key => ({
      label: getString(map[key]),
      value: key
    }))
  }

  return (
    <>
      <FormInput.MultiSelect
        name="users"
        key="users"
        items={users}
        label={getString('common.user')}
        multiSelectProps={{
          allowCreatingNewItems: true,
          onQueryChange: setUserQuery,
          tagRenderer: UserTagRenderer,
          itemRender: UserItemRenderer
        }}
      />
      {!orgIdentifier && (
        <FormInput.MultiSelect
          name="organizations"
          key="organizations"
          items={organizations}
          label="Org"
          multiSelectProps={{
            onQueryChange: setOrgQuery
          }}
        />
      )}

      {/* <ProjectsMultiSelect formikProps={props.formikProps} items={projectData?.data?.content || []} /> */}

      {/* {!projectIdentifier && (
        <FormInput.MultiSelect
          name="projects"
          key="projects"
          items={projects}
          label="Project" // add in string.yaml
          multiSelectProps={{
            onQueryChange: query => {
              setProjectsQuery(query)
              refetchProjectList()
            }
          }}
          onChange={options => {
            const orgs = options.map(option => {
              const project = projectData?.data?.content?.find(
                project => project.projectResponse.project.identifier === option.value
              )
              return {
                label: project?.organization?.name,
                value: project?.organization?.identifier
              }
            })

            props.formikProps.setFieldValue(
              'organizations',
              uniqBy(orgs, org => org.value)
            )
          }}
        />
      )} */}
      <FormInput.MultiSelect
        items={getOptionsForMultiSelect(moduleToLabelMap)}
        name="modules"
        label={getString('module')}
        key="modules"
      />
      <FormInput.MultiSelect
        items={getResourceTypeForMultiselect()}
        name="resourceType"
        label={getString('common.resourceType')}
        key="resourceType"
      />
      <FormInput.MultiSelect
        items={getOptionsForMultiSelect(actionToLabelMap)}
        name="actions"
        label={getString('action')}
        key="actions"
      />
    </>
  )
}

export default AuditTrailFilterForm
