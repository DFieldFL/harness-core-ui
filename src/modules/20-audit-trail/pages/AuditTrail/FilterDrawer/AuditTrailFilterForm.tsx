import { FormInput } from '@wings-software/uicore'
import React from 'react'
import { useStrings } from 'framework/strings'

const AuditTrailFilterForm: React.FC = () => {
  const { getString } = useStrings()
  return (
    <>
      <FormInput.Text name="action" label={getString('action')} key="action" />
      <FormInput.Text name="module" label={getString('module')} key="module" />
    </>
  )
}

export default AuditTrailFilterForm
