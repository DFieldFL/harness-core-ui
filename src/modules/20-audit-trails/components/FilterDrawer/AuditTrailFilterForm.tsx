import { FormInput, MultiSelectOption } from '@wings-software/uicore'
import React, { useState } from 'react'
import { useParams } from 'react-router-dom'
import { useMutateAsGet } from '@common/hooks'
import { StringKeys, useStrings } from 'framework/strings'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { useGetUsers, useGetOrganizationAggregateDTOList } from 'services/cd-ng'
import { UserItemRenderer, UserTagRenderer } from '@rbac/utils/utils'
import { actionToLabelMap, moduleToLabelMap } from '../../utils/RequestUtil'

const AuditTrailFilterForm: React.FC = () => {
  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
  const [userQuery, setUserQuery] = useState<string>()
  const [orgQuery, setOrgQuery] = useState<string>()

  const { data: userData } = useMutateAsGet(useGetUsers, {
    queryParams: { accountIdentifier: accountId, orgIdentifier, projectIdentifier },
    body: {
      searchTerm: userQuery
    },
    debounce: 300
  })

  const { data } = useGetOrganizationAggregateDTOList({
    queryParams: { accountIdentifier: accountId, searchTerm: orgQuery },
    debounce: 300
  })

  const { getString } = useStrings()
  const users =
    userData?.data?.content?.map(user => {
      return {
        label: user.name || user.email,
        value: user.email
      }
    }) || []

  const organizations =
    data?.data?.content?.map(org => {
      return {
        label: org.organizationResponse.organization.name,
        value: org.organizationResponse.organization.identifier
      }
    }) || []

  const getOptionsForMultiSelect = (map: Record<any, StringKeys>): MultiSelectOption[] => {
    return Object.keys(map).map(key => ({
      label: getString(map[key]),
      value: key
    }))
  }

  return (
    <>
      <FormInput.MultiSelect
        name="users"
        key="users"
        items={users}
        label={getString('common.user')}
        multiSelectProps={{
          allowCreatingNewItems: true,
          onQueryChange: setUserQuery,
          tagRenderer: UserTagRenderer,
          itemRender: UserItemRenderer
        }}
      />
      <FormInput.MultiSelect
        name="organizations"
        key="organizations"
        items={organizations}
        label="Org"
        multiSelectProps={{
          onQueryChange: setOrgQuery
        }}
      />
      <FormInput.MultiSelect
        items={getOptionsForMultiSelect(moduleToLabelMap)}
        name="modules"
        label={getString('module')}
        key="modules"
      />
      <FormInput.MultiSelect
        items={getOptionsForMultiSelect(actionToLabelMap)}
        name="actions"
        label={getString('action')}
        key="actions"
      />
    </>
  )
}

export default AuditTrailFilterForm
