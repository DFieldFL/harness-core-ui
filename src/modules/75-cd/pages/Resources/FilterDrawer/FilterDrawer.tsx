import React from 'react'
import { Drawer, IDrawerProps, Classes } from '@blueprintjs/core'
import { Button, Layout } from '@wings-software/uikit'
import i18n from './FilterDrawer.i18n'
import css from './FilterDrawer.module.scss'

class FilterDrawer extends React.Component<{ isFilterPanelOpen: boolean; onClose?: () => void }> {
  state: IDrawerProps = {
    autoFocus: true,
    canEscapeKeyClose: true,
    canOutsideClickClose: true,
    enforceFocus: true,
    hasBackdrop: true,
    usePortal: true,
    isOpen: false,
    size: 550
  }

  render() {
    return (
      <Drawer
        className={css.filterDrawer}
        {...this.state}
        isOpen={this.props.isFilterPanelOpen}
        onClose={this.props.onClose}
        icon="settings"
        title={i18n.filterTitle}
      >
        <div className={Classes.DRAWER_BODY}>
          <div className={Classes.DIALOG_BODY}>{this.props.children}</div>
        </div>
        <div className={Classes.DRAWER_FOOTER}>
          <Layout.Horizontal spacing="medium" flex={true} style={{ justifyContent: 'flex-end' }}>
            <Button text={i18n.saveFilter} intent="none" className={css.secondaryBtn} />
            <Button text={i18n.applyFilter} intent="primary" />
          </Layout.Horizontal>
        </div>
      </Drawer>
    )
  }
}

export default FilterDrawer
