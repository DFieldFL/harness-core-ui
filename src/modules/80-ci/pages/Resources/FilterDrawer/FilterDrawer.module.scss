.filterDrawer {
  :global {
    .bp3-drawer-header {
      padding: var(--spacing-large);
      box-shadow: 0 1px 0px var(--grey-200);
      [icon='settings'] {
        transform: rotate(90deg);
      }
      [data-icon='small-cross'] {
        height: 25px;
        width: 25px;
      }
      .bp3-dialog-close-button {
        &:hover {
          background: transparent;
        }
      }
    }
    .bp3-drawer-footer {
      box-shadow: none;
      padding: var(--spacing-xlarge);
      .bp3-button {
        border-radius: 6px;
        padding: 4px 18px;
        height: 28px;
      }
    }
  }

  .secondaryBtn {
    border: 1px solid var(--grey-200) !important;
    background: var(--grey-200) !important;
    font-weight: bold !important;
    color: var(--grey-800) !important;
  }
}
