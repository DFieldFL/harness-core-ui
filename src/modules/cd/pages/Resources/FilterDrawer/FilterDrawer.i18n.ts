export default {
  filterTitle: 'Filters',
  saveFilter: 'Save Filter',
  applyFilter: 'Apply'
}
