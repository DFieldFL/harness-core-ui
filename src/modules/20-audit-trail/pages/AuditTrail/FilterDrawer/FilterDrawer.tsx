import React from 'react'
import { Filter } from '@common/components/Filter/Filter'
import type { FilterDTO } from 'services/cd-ng'
import type { FilterDataInterface } from '@common/components/Filter/Constants'
import AuditTrailFilterForm from './AuditTrailFilterForm'

interface FilterDrawerProps {
  closeDrawer: () => void
}

const FilterDrawer: React.FC<FilterDrawerProps> = ({ closeDrawer }) => {
  return (
    <Filter<{}, FilterDTO>
      formFields={<AuditTrailFilterForm />}
      onApply={() => {
        //on apply filters
      }}
      onClose={closeDrawer}
      onSaveOrUpdate={(_isUpdate: boolean, _data: FilterDataInterface<{}, FilterDTO>) => {
        //empty
      }}
      onFilterSelect={() => {
        //empty
      }}
      initialFilter={{
        formValues: {},
        metadata: { filterProperties: {}, identifier: 'filterOne', name: 'filter one' }
      }}
    />
  )
}

export default FilterDrawer
