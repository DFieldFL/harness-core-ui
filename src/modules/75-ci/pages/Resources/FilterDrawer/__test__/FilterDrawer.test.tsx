import React from 'react'

import { render, queryByText } from '@testing-library/react'
import i18n from '../FilterDrawer.i18n'
import FilterDrawer from '../FilterDrawer'

describe('FilterDrawer Snapshot', () => {
  test('should render FilterDrawer', () => {
    const { container } = render(<FilterDrawer isFilterPanelOpen={true} />)
    expect(queryByText(container, i18n.filterTitle)).toBeDefined()
    expect(queryByText(container, i18n.applyFilter)).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
