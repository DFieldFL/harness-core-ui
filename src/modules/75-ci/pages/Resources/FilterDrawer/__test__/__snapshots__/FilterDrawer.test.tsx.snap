// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDrawer Snapshot should render FilterDrawer 1`] = `<div />`;
